packages/import-node-fetch/src/index.ts

```diff
@@ -0,0 +1,30 @@
+// eslint-disable-next-line @typescript-eslint/consistent-type-imports
+export type TNodeFetch = typeof import('node-fetch');
+export default async function importNodeFetch(): Promise<TNodeFetch> {
+  // Node-fetch is an ESM module from 3.x
+  // Importing ESM modules to CommonJS is possible with a dynamic import.
+  // However, once this is transpiled with TS, `await import()` changes to `require()`, which fails to load
+  // the package at runtime.
+  // The alternative, to transpile with "moduleResolution": "NodeNext", is not always feasible.
+  // Use this function to safely import the node-fetch package
+  let module;
+  try {
+    module = await import('node-fetch');
+  } catch (err: unknown) {
+    if (
+      err &&
+      typeof err === 'object' &&
+      'code' in err &&
+      err.code === 'ERR_REQUIRE_ESM'
```

Comment:
We'll want to have a `typeof __webpack_require__ === 'undefined'` check so that we don't end up trying to run a literal `import()` call if we are in a webpack environment